File: validator-app/src/main/java/org/cna/keyple/famoco/validator/rx/SchedulerProvider.java
package org.cna.keyple.famoco.validator.rx;

import javax.inject.Inject;

import io.reactivex.Scheduler;

public class SchedulerProvider {
    private Scheduler backScheduler;
    private Scheduler foreScheduler;

    @Inject
    public SchedulerProvider(Scheduler backgroundScheduler, Scheduler foregroundScheduler) {
        backScheduler = backgroundScheduler;
        foreScheduler = foregroundScheduler;
    }

    /**
     * IO thread pool scheduler
     */
    public Scheduler io() {
        return backScheduler;
    }

    /**
     * Main Thread scheduler
     */
    public Scheduler ui() {
        return foreScheduler;
    }
}

File: validator-app/src/main/kotlin/org/cna/keyple/famoco/validator/rx/SchedulerProvider.kt
package org.cna.keyple.famoco.validator.rx

import io.reactivex.Scheduler
import javax.inject.Inject

class SchedulerProvider @Inject constructor(
    private val backScheduler: Scheduler,
    private val foreScheduler: Scheduler
) {

    /**
     * IO thread pool scheduler
     */
    fun io(): Scheduler {
        return backScheduler
    }

    /**
     * Main Thread scheduler
     */
    fun ui(): Scheduler {
        return foreScheduler
    }

}